### ISSUE TITLE
NPE in `HqlQueryTransformer.isSubquery` for INSERT from SELECT HQL

### ISSUE THREAD

**Issue by kzander91** [spring-projects/spring-data-jpa#2977]
With 3.1.0, repository initialization fails with
```
Caused by: java.lang.NullPointerException: Cannot invoke "org.antlr.v4.runtime.ParserRuleContext.getParent()" because "ctx" is null
	at org.springframework.data.jpa.repository.query.HqlQueryTransformer.isSubquery(HqlQueryTransformer.java:100) ~[spring-data-jpa-3.1.0.jar:3.1.0]
```
when declaring the following named query:
```sql
insert into MyEntity (id, col)
select max(id), col
from MyEntityStaging
group by col
```

Entities:
```java
@Entity
@NamedQuery(name = "MyEntity.copyFromStaging", query = """
        insert into MyEntity (id, col)
        select max(id), col
        from MyEntityStaging
        group by col""")
class MyEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;
    private String col;
}

@Entity
class MyEntityStaging {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;
    private String col;
}
```

Repository:
```java
interface MyEntityRepository extends JpaRepository<MyEntity, Long> {
    @Modifying
    @Transactional
    void copyFromStaging();
}
```

Reproducer: 
[demo.zip](https://github.com/spring-projects/spring-data-jpa/files/11558003/demo.zip)
Extract and run `./mvnw spring-boot:run` to observe the exception.
In the _pom.xml_, change the `spring-data-jpa.version` from `3.1.0` to `3.0.6` and run again, now it works.

**Comment by gregturn** [comment]
I've reproduced the same thing using:

```
	@Test // GH-2977
	void isSubqueryThrowsException() {

		String query = """
				insert into MyEntity (id, col)
				select max(id), col
				from MyEntityStaging
				group by col
				""";

		assertThat(createQueryFor(query, Sort.unsorted())).isEqualTo(query);
	}
```

**Comment by gregturn** [comment]
That's merged to `main` and backported to `3.1.x`.

**Comment by kzander91** [comment]
That was fast, thanks a lot :)
